### PR TITLE
feat: add visit for constructor calls (#3217)

### PR DESCRIPTION
**Pull request by thomaszub**

### CODE REVIEW

File: rewrite-java/src/main/java/org/openrewrite/java/DeleteMethodArgument.java [hunk 1]
```diff
@@ -87,11 +88,22 @@ public DeleteMethodArgumentVisitor(MethodMatcher methodMatcher) {
...
+        }
+
+        @Override
+        public J.NewClass visitNewClass(J.NewClass newClass, ExecutionContext ctx) {
+            J.NewClass n = super.visitNewClass(newClass, ctx);
+            return (J.NewClass) visitMethodCall(n);
+        }
+
+        private MethodCall visitMethodCall(MethodCall methodCall) {
+            MethodCall m = methodCall;
```
Comment (timtebeek) [inline rewrite-java/src/main/java/org/openrewrite/java/DeleteMethodArgument.java:101]:
```suggestion
        private MethodCall visitMethodCall(J.MethodCall methodCall) {
            J.MethodCall m = methodCall;
```

File: rewrite-java/src/main/java/org/openrewrite/java/DeleteMethodArgument.java [hunk 2]
```diff
@@ -23,6 +23,7 @@
 import org.openrewrite.java.search.UsesMethod;
 import org.openrewrite.java.tree.Expression;
 import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.MethodCall;
```
Comment (timtebeek) [inline rewrite-java/src/main/java/org/openrewrite/java/DeleteMethodArgument.java:26]:
```suggestion
```

**Review by timtebeek — APPROVED** [review]
Looking good! Thanks for immediately providing a fix.

**Review by rpau — APPROVED** [review]
LGTM after the suggestions from Tim are addressed

File: rewrite-java/src/main/java/org/openrewrite/java/DeleteMethodArgument.java [hunk 3]
```diff
@@ -87,11 +87,22 @@ public DeleteMethodArgumentVisitor(MethodMatcher methodMatcher) {
...
+            return (J.MethodInvocation) visitMethodCall(m);
+        }
+
+        @Override
+        public J.NewClass visitNewClass(J.NewClass newClass, ExecutionContext ctx) {
+            J.NewClass n = super.visitNewClass(newClass, ctx);
+            return (J.NewClass) visitMethodCall(n);
+        }
+
+        private MethodCall visitMethodCall(J.MethodCall methodCall) {
```
Comment (timtebeek) [inline rewrite-java/src/main/java/org/openrewrite/java/DeleteMethodArgument.java:99]:
```suggestion
        private J.MethodCall visitMethodCall(J.MethodCall methodCall) {
```

**Comment by thomaszub** [comment]
@timtebeek The `MethodCall` is a separate interface not located in the `J` class. I think the suggested change would not work. This is also the reason why I added the overrides to `J.NewClass` and `J. MethodInvocation`.

**Pull request merged**